### ISSUE TITLE
Colour conversions are incorrect when converting 10bit AVIF Images to JPG

### ISSUE THREAD

**Issue by 06needhamt** [ImageMagick/ImageMagick#2415]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have verified that I am using the latest version of ImageMagick
- [X] I have searched [open](https://github.com/ImageMagick/ImageMagick/issues) and [closed](https://github.com/ImageMagick/ImageMagick/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

Colour conversions are incorrect when converting 10bit AVIF samples from [Netflix Open Content](http://download.opencontent.netflix.com/?prefix=AV1/Chimera/AVIF/)

### Steps to Reproduce

Convert `Chimera-AV1-8bit-1920x1080-6736kbps-162.avif` and `Chimera-AV1-10bit-1920x1080-6191kbps-162.avif` from the link above to jpg using the below code.

```csharp
using ImageMagick;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ImageMagickTest
{
    class Program
    {
        static void Main(string[] args)
        {
            List<string> files = Directory.EnumerateFiles(Environment.CurrentDirectory).ToList();
            foreach (string inputFile in files)
            {
                Console.WriteLine($"Converting File {Path.GetFileName(inputFile)} To .jpg");
                string outputFile = Path.GetFileNameWithoutExtension(inputFile) + ".jpg";
                bool success = ConvertImage(inputFile, outputFile);
                
                if(success)
                    Console.WriteLine($"Successfully Converted File {Path.GetFileName(inputFile)} To .jpg");
                else
                    Console.WriteLine($"Failed To Convert File {Path.GetFileName(inputFile)} To .jpg");
            }
        }

        static bool ConvertImage(string inputFile, string outputFile)
        {
            try
            {
                MagickImage image = new MagickImage(inputFile);
                int depth = image.Depth;
                image.Write(outputFile, MagickFormat.Jpg);
                image.Dispose();

                FileInfo fi = new FileInfo(outputFile);
                return fi.Exists && fi.Length > 0;
            }
            catch (Exception ex)
            {
                return false;
            }
        }
    }
}
```

Expected Behaviour:
The image should convert as below (this is the converted 8-bit sample).
![Chimera-AV1-8bit-1920x1080-6736kbps-162](https://user-images.githubusercontent.com/4162227/90232442-9a708900-de14-11ea-9fe6-06de220b2e0a.jpg)

Actual Behaviour:
The Image is converted as below
![Chimera-AV1-10bit-1920x1080-6191kbps-162](https://user-images.githubusercontent.com/4162227/90232519-b6742a80-de14-11ea-85ce-f82c0e52cc3b.jpg)

Side Note:
When debugging in Visual Studio 2019 the bit depth of the file is reported as 8-bits instead of 10-bits. 
I am unsure if this is related to the issue. See screenshot below
![image](https://user-images.githubusercontent.com/4162227/90232724-04892e00-de15-11ea-840b-1d121b29ef1d.png)

### System Configuration

- ImageMagick version: 7.0.10-25 beta
- Environment (Operating system, version and so on): Windows 10 2004 x64
- Additional information: Conversion was done using Magick.NET version 7.21.1

**Comment by snibgo** [comment]
Your linked samples directory contains many files. Which one is shown in your images?

I tried two files from that directory at the command line, using IM v7.0.10-27:

```
magick Chimera-AV1-8bit-1280x720-3363kbps-100.avif x.png```
```
That seems to work fine. Note the "8-bit" in the filename.

```
magick Chimera-AV1-10bit-1280x720-2380kbps-100.avif x.png
```
That does not work well, with a similar colour distortion to your example. Note the "10-bit" in the filename.

**Comment by dlemstra** [comment]
Will need to investigate this but I think that libheif only supports 8 bit images at the moment.

**Comment by 06needhamt** [comment]
@snibgo The images i used where Chimera-AV1-8bit-1920x1080-6736kbps-162.avif and Chimera-AV1-10bit-1920x1080-6191kbps-162.avif. I will update my post accordingly. It seems that you are getting a similar result to me with the other images.

**Comment by 06needhamt** [comment]
@dlemstra I have done a little bit of investigation and it appears that libheif supports 10-bit images. At least on the master branch. https://github.com/strukturag/libheif/blob/master/libheif/heif_decoder_aom.cc#L166

**Comment by dlemstra** [comment]
Starting with libheif 1.7.0 there is an option called convert_hdr_to_8bit added to the decoder. With this option set we get the correct output. This option will be set in the next release of ImageMagick and we will at a later moment try to get support for reading 10 bit and convert the value to the Quantum depth.

**Comment by 06needhamt** [comment]
Great thanks. Should we open a issue at libheif to add proper 10bit support or does one already exist?

**Comment by dlemstra** [comment]
I don't know if we or libheif not handling this properly. Investigated this a while back but I could not figure out why it failed.

**Comment by 06needhamt** [comment]
From what I've seen I think it's libheif's problem but I'll investigate and see what I can work out.

**Comment by 0-wiz-0** [comment]
I just tried converting https://ptrace.fefe.de/weihnachtsspielzeug.avif to png or jpg using ImageMagick 7.1-6 and libheif-1.12.0 but I get ~half the picture with black lines interlaced and it looks quite green (I don't know what the original should look like).
Is this the same issue or a separate one?

**Comment by 0-wiz-0** [comment]
Yes, firefox 93 also displays this correctly. So I think there's a bug in ImageMagick or one of the libraries it uses.

**Comment by dlemstra** [comment]
This should be fixed in the next release as reported here: #4932.
